Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -1,59 +1,48 @@
-        }
-    }
-}
-
 // MARK: - Dependencies Controlling
 
 /// `DependenciesControlling` controls:
 ///     1. Fetching/updating dependencies defined in `./Tuist/Dependencies.swift` by running appropriate dependencies managers (`Cocoapods`, `Carthage`, `SPM`).
 ///     2. Compiling fetched/updated depedencies into `.framework.`/`.xcframework.`.
-///     3. Saving compiled frameworks uder `./Tuist/Dependencies/*`.
+///     3. Saving compiled frameworks under `./Tuist/Dependencies/*`.
+///     4. Generating dependencies graph under `./Tuist/Dependencies/graph.json`.
 public protocol DependenciesControlling {
-    /// Fetches dependencies.
-    /// - Parameter path: Directory whose project's dependencies will be fetched.
-    func fetch(at path: AbsolutePath) throws
-    /// Updates dependencies.
-    /// - Parameter path: Directory whose project's dependencies will be updated.
-    func update(at path: AbsolutePath) throws
+    /// Installes dependencies.
+    /// - Parameter path: Directory whose project's dependencies will be installed.
+    /// - Parameter method: Installation method.
+    /// - Parameter dependencies: List of dependencies to intall.
+    func install(at path: AbsolutePath, method: InstallDependenciesMethod, dependencies: Dependencies) throws
 }
 
 // MARK: - Dependencies Controller
 
 public final class DependenciesController: DependenciesControlling {
-    public init() {}
+    private let carthageInteractor: CarthageInteracting
+    private let cocoaPodsInteractor: CocoaPodsInteracting
+    private let swiftPackageManagerInteractor: SwiftPackageManagerInteracting
+
+    public init(
+        carthageInteractor: CarthageInteracting = CarthageInteractor(),
+        cocoaPodsInteractor: CocoaPodsInteracting = CocoaPodsInteractor(),
+        swiftPackageManagerInteractor: SwiftPackageManagerInteracting = SwiftPackageManagerInteractor()
+    ) {
+        self.carthageInteractor = carthageInteractor
+        self.cocoaPodsInteractor = cocoaPodsInteractor
+        self.swiftPackageManagerInteractor = swiftPackageManagerInteractor
+    }
 
-    public func fetch(at _: AbsolutePath) throws {
-        logger.notice("Start fetching depednencies.")
+    public func install(at path: AbsolutePath, method: InstallDependenciesMethod, dependencies: Dependencies) throws {
+        logger.notice("Start installing Carthage dependencies.")
 
-        // TODO: implement me!
-        throw DependenciesControllerError.unimplemented
-    }
+        let dependenciesDirectory = path
+            .appending(component: Constants.tuistDirectoryName)
+            .appending(component: Constants.DependenciesDirectory.name)
 
-    public func update(at _: AbsolutePath) throws {
-        logger.notice("Start updating depednencies.")
+        try carthageInteractor.install(dependenciesDirectory: dependenciesDirectory, method: method, dependencies: dependencies.carthageDependencies)
```

Comment:
I think we can log a section right before this line saying that we are starting to update the Carthage dependencies.